lib/srv/desktop/discovery.go

```diff
@@ -122,7 +122,15 @@ func (s *WindowsService) getDesktopsFromLDAP() types.ResourcesWithLabels {
 	s.cfg.Log.Debugf("searching for desktops with LDAP filter %v", filter)
 
 	entries, err := s.lc.readWithFilter(s.cfg.DiscoveryBaseDN, filter, computerAttribtes)
-	if err != nil {
+	if trace.IsConnectionProblem(err) {
+		// If the connection was broken, re-initialize the LDAP client so that it's
+		// ready for the next reconcile loop. Return the last known set of desktops
+		// in this case, so that the reconciler doesn't delete the desktops it already
+		// knows about.
+		s.cfg.Log.Info("LDAP connection error when searching for desktops, reinitializing client")
+		s.initializeLDAP()
```

Comment:
Yes, will add some logging as Tim pointed out the same.

We're not "leaking" that goroutine, as it will be terminated when the service shuts down. We do still need to periodically renew our certificate, but I think what you're pointing out is that we can technically cancel the renewal we've already scheduled as this will schedule a new one.

I don't think there's an issue here though, as:

- there's nothing bad about renewing our certificate more frequently than necessary
- we only schedule a renewal if we successfully got a cert, so in cases where we're in an error loop we won't be scheduling a bunch of renewals.

Thoughts?

Reply:
@ibeckermayer check out the latest commit - I think it's a better way to address the renewal and should address your comment.

Reply:
> We're not "leaking" that goroutine, as it will be terminated when the service shuts down.

True, bad description of the problem on my part.

> check out the latest commit - I think it's a better way to address the renewal and should address your comment.

Yeah looks good.